connector/src/main/java/org/jboss/as/connector/subsystems/resourceadapters/Constants.java

```diff
@@ -659,6 +662,14 @@ public void marshallAsElement(AttributeDefinition attribute, ModelNode resourceM
             .setRestartAllServices()
             .build();
 
+    static SimpleAttributeDefinition REPORT_DIRECTORY = new SimpleAttributeDefinitionBuilder(REPORT_DIRECTORY_NAME, ModelType.STRING)
+            .setXmlName("path")
+            .setDefaultValue(new ModelNode(new ValueExpression("${jboss.server.log.dir}")))
+            .setRequired(false)
+            .setAllowExpression(true)
+            .setRestartAllServices()
+            .build();
```

Comment:
Please add this to the builder:

.addArbitraryDescriptor(ModelDescriptionConstants.FILESYSTEM_PATH, ModelNode.TRUE)

This provides a hint to tools like the CLI that they can use the local FS to do tab completion suggestions.

Reply:
@jmesnil or anyone... Plus @tadamski of course

This is non-standard in how it is handling paths; i.e. it violates the 2 attribute, path + relative-to convention.

OTOH I really like the default value idea.

The downside of not having relative-to is the path can't be relative to user-provided Path resources, as those don't have associated system properties. But if that is a problem we can compatibly add another report-directory-relative-to attribute later.

Note that with a 2 attribute setup, the default value Tomek added here is still valid; the 'relative-to' would have no default.

Perhaps this should be named report-directory-path to better align with a future report-directory-relative-to but I don't really think that is necessary.

@jamezp made a comment re possible effect on wildscribe that i'll look into a bit.

Reply:
Wildscribe renders the default value as '${jboss.server.log.dir}' which isn't as correct as the 'expression "${jboss.server.log.dir}"' in the read-resource-description output but seems ok.

Reply:
@tadamski Please add this to the builder:

.addArbitraryDescriptor(ModelDescriptionConstants.FILESYSTEM_PATH, ModelNode.TRUE)

Reply:
added

Reply:
Thanks. I resolved this conversation but I'm going to 'unresolve' it, not because it's not done, but just because I think the discussion comments are useful so no need to make the github ui hide them.